components/setting_item.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode, RefObject} from 'react';
+
+import SettingItemMin from 'components/setting_item_min';
+
+type Props = {
+
+    /**
+     * Whether this setting item is currently open
+     */
+    active: boolean;
+
+    /**
+     * Whether all sections in the panel are currently closed
+     */
+    areAllSectionsInactive: boolean;
+
+    /**
+     * The identifier of this section
+     */
+    section: string;
+
+    /**
+     * The setting UI when it is maximized (open)
+     */
+    max: ReactNode | null;
+
+    // Props to pass through for SettingItemMin
+    updateSection: (section: string) => void;
+    title?: ReactNode;
+    disableOpen?: boolean;
+    describe?: ReactNode;
+}
+export default class SettingItem extends React.PureComponent<Props> {
+    minRef: RefObject<typeof SettingItemMin>;
```

Comment:
Note, this line is what's causing type check to fail... I want to use `SettingItemMin` here but it's a HOC wrapped by `ConnectComponent`. I added `typeof` to satisfy the IDE but that's obviously not correct. Any suggestions? 🤔 I don't want to have to rewrite `SettingItemMin` as a functional component to take a forwardRef.

Reply:
Resolved, I just needed to use the type of the actual component rather than the wrapped one.